src/Framework/DomQuery.php

```diff
@@ -24,6 +24,9 @@ public static function fromHtml($html)
 		}
 
 		$html = preg_replace('#<(keygen|source|track|wbr)(?=\s|>)("[^"]*"|\'[^\']*\'|[^"\'>]+)*+(?<!/)>#', '<$1$2 />', $html);
+		if (LIBXML_VERSION < 20800) {
+			$html = preg_replace('~\\<meta\\s+.*?charset=(?:"|\')(.+)(?:"|\').*?\\>~i', '<meta http-equiv="Content-Type" content="text/html; charset=$1" />', $html);
```

Comment:
This will *eat* `<meta xyz> charset="utf-8" <b>` and doesn't support `<meta charset=utf-8>`.